torch/csrc/distributed/rpc/rref_impl.h

```diff
@@ -395,6 +399,28 @@ class TORCH_API OwnerRRef final : public RRef {
   friend class RRefContext;
 
   std::shared_ptr<JitFuture> future_;
+
+ public:
+  // Records an event per each stream in the context and stores them in
+  // the current OwnerRRef instance if CUDA is available,
+  // otherwise does nothing.
+  void recordAllStreams(std::shared_ptr<LazyStreamContext> ctx);
+
+  // Blocks all streams in the context on all events previously stored in
+  // the current OwnerRRef instance if CUDA is available,
+  // otherwise does nothing.
+  void blockAllStreams(std::shared_ptr<LazyStreamContext> ctx);
+ private:
+
+ #ifdef USE_CUDA_NOT_ROCM
```

Comment:
Let's continue discussing this.

> Curious, why the different class sizes could be a problem? I thought these are compile time flags, which means when CUDA is not available, it's same as this class does not have this field, no?

> The practice shows that this header is included to cpp files with CUDA and without and it produces segfaults

Sorry, I still didn't follow. Does this mean we need to propagate `USE_CUDA` properly to those files as well? Would I be correct if I assume those files are the non `tensorpipe_*` ones in `libtorch_python_distributed_sources`? E.g., the `py_rref.h` below?

https://github.com/pytorch/pytorch/blob/305abde976d232494c6b5252111ce15523511ffe/torch/csrc/distributed/rpc/py_rref.h#L3

If so, does it mean we should propagate `USE_CUDA` to those files in the following location?

https://github.com/pytorch/pytorch/blob/305abde976d232494c6b5252111ce15523511ffe/caffe2/CMakeLists.txt#L376-L385